contracts/minter/src/contract.rs

```diff
@@ -440,17 +455,21 @@ fn _execute_mint(
     // remove mintable token id from map
     MINTABLE_TOKEN_IDS.remove(deps.storage, mintable_token_id);
 
-    let seller_amount = mint_price.amount - network_fee;
-    let seller_msg = BankMsg::Send {
-        to_address: config.admin.to_string(),
-        amount: vec![coin(seller_amount.u128(), config.unit_price.denom)],
+    let seller_msg: Vec<BankMsg> = if admin_no_fee {
+        vec![]
+    } else {
+        let seller_amount = mint_price.amount - network_fee;
+        vec![BankMsg::Send {
+            to_address: config.admin.to_string(),
+            amount: vec![coin(seller_amount.u128(), config.unit_price.denom)],
+        }]
     };
-    msgs.append(&mut vec![seller_msg.into()]);
 
     Ok(Response::default()
         .add_attribute("action", action)
         .add_messages(msgs)
-        .add_messages(network_fee_msg))
+        .add_messages(network_fee_msgs)
```

Comment:
So instead of sending empty messages here, another approach is to create a mutable messages array, and just append the relevant messages to it.

Reply:
i like it